### PR TITLE
Connection mail notification

### PR DESCRIPTION
**Pull request by rifeman2007**

### CODE REVIEW

File: src/AppBundle/Command/ConnectionFollowUpEmailCommand.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $days       = $input->getArgument('days');
+        $jobManager = $this->getContainer()->get('job_manager');
+        $jobType    = ($days == Connection::FOLLOW_UP_14_DAYS? Job::CONNECTION_MAIL_FOLLOW_UP_14: Job::CONNECTION_MAIL_FOLLOW_UP_42);
+
+        if ($jobManager->isRunning($jobType)) {
```
Comment (jongotlin) [inline src/AppBundle/Command/ConnectionFollowUpEmailCommand.php]: Use the LockHandler instead http://symfony.com/doc/current/components/filesystem/lock_handler.html

File: src/AppBundle/Command/ConnectionFollowUpEmailCommand.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            $output->writeln('Connection follow-up mail job is still running.');
+
+            return;
+        }
+
+        gc_enable();
```
Comment (jongotlin) [inline src/AppBundle/Command/ConnectionFollowUpEmailCommand.php]: Do you really need to manually handle gc?

File: src/AppBundle/Command/ConnectionFollowUpEmailCommand.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        $output->writeln('Connection follow-up mail job has started.');
+
+        $ctr                = 0;
+        $numSent            = ($days == Connection::FOLLOW_UP_14_DAYS? 0: 1);
+        $templating         = $this->getContainer()->get('templating');
+        $connectionManager  = $this->getContainer()->get('connection_manager');
+        $userMailer         = $this->getContainer()->get('app.user_mailer');
+
+        try {
+            while ((($connection = $connectionManager->getFindCreatedDaysAgo($days, $numSent)) instanceof Connection)) {
```
Comment (jongotlin) [inline src/AppBundle/Command/ConnectionFollowUpEmailCommand.php]: Please use foreach-syntax.

File: src/AppBundle/Entity/Connection.php [hunk 4]
```diff
@@ -87,11 +91,19 @@ class Connection
      */
     protected $musicFriend = false;
 
+    /**
+     * @var boolean
+     *
+     * @ORM\Column(type="integer")
+     */
+    protected $numSent;
```
Comment (jongotlin) [inline src/AppBundle/Entity/Connection.php]: This is not a very descriptive name. `followUpEmailsSentCount`?

File: src/AppBundle/Entity/ConnectionRepository.php [hunk 5]
```diff
@@ -153,4 +153,24 @@ public function isUserConnectionExists(User $user1, User $user2)
...
+    /**
+     * @param $days
+     * @param $numSent
+     * @return array
+     */
+    public function findCreatedDaysAgo($days, $numSent)
+    {
+        return $this
+            ->createQueryBuilder('c')
+            ->where('DATE_DIFF(:now, c.createdAt) >= :days')
```
Comment (jongotlin) [inline src/AppBundle/Entity/ConnectionRepository.php]: With the `>=`-operator we will send emails to __all__ connections ever made when this is run for the first time. Is this by design @mikaelohren?

Comment (mikaelohren) [inline src/AppBundle/Entity/ConnectionRepository.php]: @jongotlin no this is not a good idea. I do not want all connections ever made to get this email. @rifeman2007 what is you thought on this?

File: src/AppBundle/Command/ConnectionFollowUpEmailCommand.php (hunk 2 above)
Comment (mikaelohren) [inline src/AppBundle/Command/ConnectionFollowUpEmailCommand.php]: @rifeman2007 ?

**Pull request closed without merging**